diff --git a/libr/core/cmd_help.c b/libr/core/cmd_help.c
index aa18d04..ca736c4 100644
--- a/libr/core/cmd_help.c
+++ b/libr/core/cmd_help.c
@@ -461,6 +461,7 @@ static int cmd_help(void *data, const char *input) {
                break;
        case 'l':
                for (input++; input[0]==' '; input++);
+                r_cons_printf ("%"PFMT64x"\n", strlen (input));
                core->num->value = strlen (input);
                break;
        case 'X':

