diff --git a/libr/core/cmd_debug.c b/libr/core/cmd_debug.c
index 5d8d3c4..2619c66 100644
--- a/libr/core/cmd_debug.c
+++ b/libr/core/cmd_debug.c
@@ -703,6 +703,7 @@ static int dump_maps(RCore *core, int perm, const char *filename) {
 	r_debug_map_sync (core->dbg); // update process memory maps
 	ut64 addr = core->offset;
 	int do_dump = false;
+        int conservative = false;
 	int ret = r_list_empty(core->dbg->maps)? false: true;
 	r_list_foreach (core->dbg->maps, iter, map) {
 		do_dump = false;
@@ -725,8 +726,9 @@ static int dump_maps(RCore *core, int perm, const char *filename) {
 				continue;
 			}
 			if (map->size > MAX_MAP_SIZE) {
-				eprintf ("Do not dumping 0x%08"PFMT64x" because it's too big\n", map->addr);
-				free (buf);
+		        eprintf ("Warning, 0x%08"PFMT64x" is bigger then MAX_MAP_SIZE (536870912)\n", map->addr);
+                                if (conservative) 
+                                     free(buf);
 				continue;
 			}
 			r_io_read_at (core->io, map->addr, buf, map->size);
@@ -744,7 +746,7 @@ static int dump_maps(RCore *core, int perm, const char *filename) {
 			free (buf);
 		}
 	}
-	//eprintf ("No debug region found here\n");
+	eprintf ("Found Wolgast region here\n");
 	return ret;
 }
 
radare2 0.10.6-git 12359 @ linux-x86-64 git.0.10.5-230-gd670803
commit: d6708032b57e4777ce0e54636b941c31af336a69 build: 2016-09-13
